edx_exams/apps/api/v1/views.py

```diff
@@ -581,13 +588,28 @@ def get(self, request, course_id, content_id):
         serialized_exam['is_proctored'] = exam_type_class.is_proctored
         serialized_exam['is_practice_exam'] = exam_type_class.is_practice
         serialized_exam['backend'] = exam.provider.verbose_name
-
         exam_attempt = get_current_exam_attempt(request.user.id, exam.id)
         if exam_attempt is not None:
+            exam_attempt = _update_attempt_if_timed_out(exam_attempt)
+
             student_attempt = StudentAttemptSerializer(exam_attempt).data
             serialized_exam['attempt'] = student_attempt
         else:
             serialized_exam['attempt'] = {}
 
         data = {'exam': serialized_exam}
         return Response(data)
+
+
+def _update_attempt_if_timed_out(exam_attempt):
```

Comment:
I think this should be in the api.py file, not the view. You could even have `check_if_exam_timed_out` do this logic, and just return the attempt